### ISSUE TITLE
Provide a way to cancel and reschedule tasks

### ISSUE THREAD

**Issue by tedsteen** [kagkarlsson/db-scheduler#4]
Make it possible to cancel and reschedule given a task name and an instance Id.
So basically something like this:
```
final String taskName = "myTaskName";
final OneTimeTask task = new MyOneTimeTask(taskName);
final String instanceId = "myInstanceId";

scheduler.scheduleForExecution(Instant.now().plusSeconds(5), task.instance(instanceId));

scheduler.reSchedule(taskName, instanceId, Instant.now().plusSeconds(15));
//Or
scheduler.cancelSchedule(taskName, instanceId);
```

**Comment by kagkarlsson** [comment]
Excellent suggestion :). Will try to add support for this in the coming week.

**Comment by tedsteen** [comment]
>Excellent suggestion :).

Thanks!
>Will try to add support for this in the coming week.

If you could give me a few hints how to go about this (simple list of basic steps needed to achieve this) I could make a PR for you to approve or continue working on.

**Comment by kagkarlsson** [comment]
I think we need to 
* Add the methods to `SchedulerClient` interface, and implement in `Scheduler`
* Easiest thing is to use existing methods for reschedule and remove in JdbcTaskRepository, after using getExecution to find any execution matching task+instance

**Comment by kagkarlsson** [comment]
Merged PR to master. Will hopefully get to do a new release soon, just have to update some docs first.

**Comment by kagkarlsson** [comment]
This feature was implemented in version 2.0 thanks to the PR from @tedsteen .
Closing the issue.
